### PR TITLE
Templating Dockerfiles - NODE/PYTHON

### PR DESCRIPTION
**Pull request by CooperLink**
[DO NOT MERGE] PR for tracking ... still templating/testing python 

Follow up PR for https://github.com/Azure/azure-functions-docker/pull/516  to bring changes to node and python images. 
Testing in progress for Python images. Node has been tested. 

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### CODE REVIEW

**Comment by CooperLink** [comment]
Need to understand why /out/ was not .gitignored and delete those files from this pull

File: host/3.0/buster/amd64/out/java/java11-appservice.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,70 @@
+ARG JAVA_VERSION=11u7
+
+ARG HOST_VERSION=3.3.1
+
+FROM mcr.microsoft.com/dotnet/sdk:5.0 AS runtime-image
+ARG HOST_VERSION
+
+ENV PublishWithAspNetCoreTargetManifest=false
+ENV DEBIAN_FRONTEND=noninteractive
```
Comment (pragnagopa) [inline host/3.0/buster/amd64/out/java/java11-appservice.Dockerfile]: delete changes in Java files? PR should be scoped to Node / Python

Comment (CooperLink) [reply host/3.0/buster/amd64/out/java/java11-appservice.Dockerfile]: Yes, this is related to the thing I mentioned before. For whatever reason, git did not follow the gitignore file that I added in the last PR. These out/ files should not be pushed

File: host/3.0/node-build.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ pr:
       - release/3.x
   paths:
     include:
+      - host/3.0/buster/amd64/base/*
```
Comment (pragnagopa) [inline host/3.0/node-build.yml:11]: Duplicate line?

**Review by pragnagopa — APPROVED** [review]

**Pull request merged**